### ISSUE TITLE
Rename to overwrite

### ISSUE THREAD

**Issue by thany** [Microsoft/vscode#70261]
As a user, I want to rename a file to a filename that already exists, overwriting that file, so that I can quickly replace an old file with a new one just by renaming.

Current behaviour: renaming a file to a name that already exists, results in an error telling me to enter a different filename.

Example use-case:
I'm developing a theme for some CMS, and in that theme I am required to include a screenshot. The screenshot is called `theme.png`. Now that I've updated the theme, I also need to update the screenshot. So I take a screenshot with my browser, which produces a filename like `Screenshot_The title of the website.png`. I then drag that file into my project, and try renaming it to `theme.png`, only to discover I need to manually delete the old file first. Only then I am allowed try the rename again.

This already works:
In Total Commander! If you rename a file in Total Commander, and the new filename you entered already exists, you get a prompt asking to overwrite. This makes perfect sense to me. Because you're essentially moving a file "to a new name" when renaming. Moving a file in vscode *does* allow the user to decide to overwrite, so by this logic renaming should, too.

**Comment by thany** [comment]
I disagree. File handling is basic stuff. Delegating too much functionality to extensions is also not desirable. I don't want to instal 15000 extensions to get a program to where I want.

On top of that, this is functionality that isn't in anyone's way. If you never rename a file, you won't need this. If you do, you're pleasantly surprized because overwriting an existing file is just being helpful.

Please reconsider.

**Comment by isidorn** [comment]
I can reopen this but I know we will never do this.
VSCode is simply designed in such a way that it relies heavily on extensions. And this is simply one of those things which shuold not be in core to not increase bloat.

**Comment by thany** [comment]
Let me repeat myself:

Overwriting is already asked when moving/copying a file. Renaming should do the same.

Why do you ask for overwrite when moving/copying a file in the first place? By you own logic, you should leave that to extensions. Why is this particular overwrite question in the core, and my scenario is brushed off? How does that make sense? What is the fundamental difference?

**Comment by isidorn** [comment]
Sure, I can reopen. I see your logic.

**Comment by thany** [comment]
That's too bad, since it's so logical to ask for overwrite instead of speweing out an error. And currently this is inconsistent between copying and renaming.

**Comment by solegaonkar** [comment]
Looks like this is resolved now.. https://code.visualstudio.com/updates/v1_72#_disable-incremental-naming
